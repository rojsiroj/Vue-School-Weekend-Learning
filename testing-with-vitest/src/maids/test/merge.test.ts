import { test, expect } from 'vitest'
import { deepMerge } from '..'

test('shallow merge', () => {
  const merged = deepMerge({ name: 'Siroj' }, { github: 'rojsiroj' })

  expect(merged).toEqual({
    name: 'Siroj',
    github: 'rojsiroj'
  })
})

test('shallow merge with overlaps', () => {
  const merged = deepMerge({ name: 'Siroj', github: 'rojsiroj' }, { twitter: 'rojsiroj_' })

  expect(merged).toEqual({
    name: 'Siroj',
    github: 'rojsiroj',
    twitter: 'rojsiroj_'
  })
})

test('deep merge with overlaps', () => {
  const merged = deepMerge(
    { name: 'Siroj', accounts: { github: 'rojsiroj' }, languages: ['javascript'] },
    { accounts: { twitter: 'rojsiroj_' }, languages: ['typescript', 'vue'] }
  )

  expect(merged).toMatchSnapshot()
})

test('shallow merge with arrays', () => {
  const merged = deepMerge(['vue', 'react'], ['svelte', 'solid'])

  expect(merged).toEqual(['vue', 'react', 'svelte', 'solid'])
})

test('throws errors on merging to different types', () => {
  expect(() => deepMerge(['foo', 'bar'], { foo: 'bar' })).toThrowError(
    'cannot merge two different types'
  )
})
