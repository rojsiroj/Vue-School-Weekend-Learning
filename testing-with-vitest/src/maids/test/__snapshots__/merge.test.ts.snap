// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`deep merge with overlaps 1`] = `
{
  "accounts": {
    "github": "rojsiroj",
    "twitter": "rojsiroj_",
  },
  "languages": [
    "javascript",
    "typescript",
    "vue",
  ],
  "name": "Siroj",
}
`;
